src/aspire/nufft/finufft.py

```diff
@@ -130,7 +133,9 @@ def adjoint(self, signal):
                 signal = signal.reshape(self.num_pts)
 
         # FINUFFT was designed for a complex input array
-        signal = np.array(signal, copy=False, dtype=self.complex_dtype, order="C")
+        signal = np.array(
+            signal, copy=COPY_ME_MAYBE, dtype=self.complex_dtype, order="C"
```

Comment:
This one can also use `asarray()` and removing the `copy=False` completely.  As to `astype`: `copy=False` wasn't changed for it, precisely because it seems like unnecessary churn.

For the `sqrt`/weight change.  It might make sense to consider using `weight = float(weight)` if you know that is always correct (longdouble might be a problem).

Reply:
indeed, thanks!